### PR TITLE
Fix -new inspector- Inconsistencies with exported enums - Fixes remai…

### PR DESCRIPTION
**Pull request by Douglas-W-Williams**
…ning issues in #19534 and #15585

### CODE REVIEW

**Comment by rminderhoud** [comment]
Would love for this to get merged. Currently it's impossible to use non sequential enum values as an export variable.

**Comment by rminderhoud** [comment]
@ordigdug Actually it seems something is not working 100%. Create a blank scene then add a camera OR load a scene with a camera node and try selecting the camera node:

```
ERROR: operator[]: FATAL: Index p_index=1 out of size (size()=1)
   At: core/vector.h:138.
handle_crash: Program crashed with signal 4
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x37140) [0x7f403ddaf140] (??:0)
[2] Vector<String>::operator[](int) (??:0)
[3] EditorPropertyEnum::update_property() (??:0)
[4] EditorInspector::update_tree() (??:0)
[5] EditorInspector::edit(Object*) (??:0)
[6] EditorNode::_edit_current() (??:0)
[7] EditorNode::push_item(Object*, String const&, bool) (??:0)
[8] SceneTreeDock::_create() (??:0)
[9] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[10] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (??:0)
[11] Object::emit_signal(StringName const&, Variant const**, int) (??:0)
[12] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[13] CreateDialog::_confirmed() (??:0)
[14] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[15] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (??:0)
[16] Object::emit_signal(StringName const&, Variant const**, int) (??:0)
[17] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[18] Tree::propagate_mouse_event(Point2i const&, int, int, bool, TreeItem*, int, Ref<InputEventWithModifiers> const&) (??:0)
[19] Tree::propagate_mouse_event(Point2i const&, int, int, bool, TreeItem*, int, Ref<InputEventWithModifiers> const&) (??:0)
[20] Tree::propagate_mouse_event(Point2i const&, int, int, bool, TreeItem*, int, Ref<InputEventWithModifiers> const&) (??:0)
[21] Tree::_gui_input(Ref<InputEvent>) (??:0)
[22] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[23] Object::call_multilevel(StringName const&, Variant const**, int) (??:0)
[24] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[25] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (??:0)
[26] Viewport::_gui_input_event(Ref<InputEvent>) (??:0)
[27] Viewport::input(Ref<InputEvent> const&) (??:0)
[28] Viewport::_vp_input(Ref<InputEvent> const&) (??:0)
[29] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[30] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (??:0)
[31] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[32] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[33] SceneTree::input_event(Ref<InputEvent> const&) (??:0)
[34] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (??:0)
[35] InputDefault::parse_input_event(Ref<InputEvent> const&) (??:0)
[36] OS_X11::process_xevents() (??:0)
[37] OS_X11::run() (??:0)
[38] /mnt/blue/code/godot/bin/godot.x11.tools.64(main+0xd5) [0x5596863a617f] (??:0)
[39] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1) [0x7f403dd991c1] (??:0)
[40] /mnt/blue/code/godot/bin/godot.x11.tools.64(_start+0x2a) [0x5596863a5fca] (??:0)
-- END OF BACKTRACE --
```

**Comment by ordigdug** [comment]
@rminderhoud I see what check I forgot, will fix and update pull request.

**Comment by ordigdug** [comment]
@rminderhoud Updated pull request. Please test.

**Comment by rminderhoud** [comment]
Build error locally and CI also reporting build errors

```
[100%] editor/editor_properties.cpp: In member function 'void EditorPropertyEnum::_option_selected(int)':
editor/editor_properties.cpp:396:63: error: invalid use of non-static member function 'StringName EditorProperty::get_edited_property()'
   emit_signal("property_changed", get_edited_property, p_which);
                                                               ^
In file included from ./editor/editor_plugin.h:34:0,
                 from ./editor/editor_help.h:34,
                 from ./editor/create_dialog.h:34,
                 from editor/editor_properties.h:34,
                 from editor/editor_properties.cpp:31:
./editor/editor_inspector.h:101:13: note: declared here
  StringName get_edited_property();
             ^~~~~~~~~~~~~~~~~~~
scons: *** [editor/editor_properties.x11.tools.64.o] Error 1
scons: building terminated because of errors.
```

**Comment by ordigdug** [comment]
Sorry typo

**Comment by rminderhoud** [comment]
:+1: works for me

**Comment by rminderhoud** [comment]
Found another issue:

- Create a mesh instance + mesh shape
- Create new spatial material material
- Changing `parameter->cullmode` causes a crash (other parameters also affected)

Backtrace
```shell
~$ ERROR: operator[]: FATAL: Index p_index=1 out of size (size()=1)
   At: core/vector.h:138.
handle_crash: Program crashed with signal 4
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x37140) [0x7f5f7105c140] (??:0)
[2] Vector<String>::operator[](int) (??:0)
[3] EditorPropertyEnum::_option_selected(int) (??:0)
[4] MethodBind1<int>::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[5] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (??:0)
[6] Object::emit_signal(StringName const&, Variant const**, int) (??:0)
[7] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[8] OptionButton::_select(int, bool) (??:0)
[9] OptionButton::_selected(int) (??:0)
[10] MethodBind1<int>::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[11] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (??:0)
[12] Object::emit_signal(StringName const&, Variant const**, int) (??:0)
[13] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[14] PopupMenu::activate_item(int) (??:0)
[15] PopupMenu::_gui_input(Ref<InputEvent> const&) (??:0)
[16] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[17] Object::call_multilevel(StringName const&, Variant const**, int) (??:0)
[18] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[19] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (??:0)
[20] Viewport::_gui_input_event(Ref<InputEvent>) (??:0)
[21] Viewport::input(Ref<InputEvent> const&) (??:0)
[22] Viewport::_vp_input(Ref<InputEvent> const&) (??:0)
[23] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (??:0)
[24] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (??:0)
[25] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[26] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (??:0)
[27] SceneTree::input_event(Ref<InputEvent> const&) (??:0)
[28] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (??:0)
[29] InputDefault::parse_input_event(Ref<InputEvent> const&) (??:0)
[30] OS_X11::process_xevents() (??:0)
[31] OS_X11::run() (??:0)
[32] /mnt/blue/code/godot/bin/godot.x11.tools.64(main+0xd5) [0x5560479e417f] (??:0)
[33] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1) [0x7f5f710461c1] (??:0)
```

File: editor/editor_properties.cpp [hunk 1]
```diff
@@ -390,13 +390,36 @@ EditorPropertyCheck::EditorPropertyCheck() {
 
 void EditorPropertyEnum::_option_selected(int p_which) {
 
-	emit_signal("property_changed", get_edited_property(), p_which);
+	String text = options->get_item_text(p_which);
+	Vector<String> text_split = text.split(":");
+	if (text_split.size() == 1) {
+		emit_signal("property_changed", get_edited_property(), p_which);
```
Comment (rminderhoud) [inline editor/editor_properties.cpp]: Did you miss a return after this statement?

Comment (ordigdug) [reply editor/editor_properties.cpp]: Yeah, thanks for catching that. I really shouldn't code at the end of a long day. I'll fix it.

**Review by rminderhoud — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**